### DISCUSSION TITLE
Undefined global `Snacks`

### DISCUSSION THREAD

**Discussion by swarn** [folke/snacks.nvim#242] [Q&A] [answered]
When I write e.g.
```lua
  vim.api.nvim_create_user_command(
    "Bdelete",
    function() Snacks.bufdelete() end,
    { desc = "Delete buffer without changing windows" }
  )
```
LuaLS flags `Snacks` as an undefined global. Do I need to update my `lazydev` config to let it know about this global?

**Comment by folke** [comment, marked as answer]
add `---@module 'snacks'`

**Reply by swarn** [reply]
Thanks!

**Reply by wigging** [reply]
I tried this but I still get the undefined global Snacks warning. The keymap works so I guess Snacks is found but it would be nice if I could get rid of the warning. Is there a certain place in my Neovim config where I need to define the `@module` statement?

```lua
---@module 'snacks'
vim.keymap.set("n", "<leader>bd", function() Snacks.bufdelete() end)
```

**Reply by swarn** [reply]
@wigging , are you using [lazydev](https://github.com/folke/lazydev.nvim)? It configures luaLS while you're editing your config files; otherwise, you need to do something else so that luaLS knows where to find your plugins.

If you _are_ using lazydev, you can set it up so that it recognizes when you use the `Snacks` global and automatically brings in the `snacks` module:
```lua
--- @type LazyPluginSpec
local M = { "folke/lazydev.nvim" }

M.opts = {
  library = {
    { path = "snacks.nvim", words = { "Snacks" } },
  },
}

return M
```

**Reply by wigging** [reply]
No, I am not using lazydev. My LSP config is:
```lua
return {
  {
    "williamboman/mason.nvim",
    lazy = false,
    config = function()
      require("mason").setup()
    end
  },
  {
    "williamboman/mason-lspconfig.nvim",
    lazy = false,
    config = function()
      require("mason-lspconfig").setup({
        ensure_installed = { "lua_ls", "pyright" }
      })
    end
  },
  {
    "neovim/nvim-lspconfig",
    lazy = false,
    config = function()
      local lspconfig = require("lspconfig")
      lspconfig.lua_ls.setup({
        settings = {
          Lua = {
            diagnostics = { globals = { "vim", "Snacks" } },
            telemetry = { enable = false }
          }
        }
      })
      lspconfig.pyright.setup({})
    end
  }
}
```

However, with this configuration, I can add Snacks to the diagnostics setting which fixes the warning.

```lua
diagnostics = { globals = { "vim", "Snacks" } },
```
